Gems/SimulationInterfacesROS2/Code/Tests/Tools/InterfacesTest.cpp

```diff
@@ -0,0 +1,409 @@
+                        EXPECT_EQ(boxShape->m_dimensions.GetZ(), dims.GetZ());
+
+                    }
+                    auto loc = filter.m_bounds_pose.GetTranslation();
+                    EXPECT_EQ(loc, AZ::Vector3::CreateZero());
+                    return AZ::Success(EntityNameList{ "FooBar" });
+                }));
+
+        auto future = client->async_send_request(request);
+        SpinAppSome();
+
+        ASSERT_TRUE(future.wait_for(std::chrono::seconds(0)) == std::future_status::ready) << "Service call timed out.";
+        auto response = future.get();
+        EXPECT_EQ(response->result.result, simulation_interfaces::msg::Result::RESULT_OK);
+    }
+
+    //! Try to call the service with invalid data (first point is greater than second point in box)
+    TEST_F(SimulationInterfaceROS2TestFixture, GetEntitiesWithShapeBoxInvalid)
+    {
+         using ::testing::_;
+        auto node = GetRos2Node();
+
+        // strict mock, since we don't want to call the real implementation in this test
+        [[maybe_unused]] testing::StrictMock<SimulationEntityManagerMockedHandler> mock;
+
+        auto client = node->create_client<simulation_interfaces::srv::GetEntities>("/get_entities");
+        auto request = std::make_shared<simulation_interfaces::srv::GetEntities::Request>();
+        request->filters.bounds.points.resize(2);
+
+        request->filters.bounds.points[0].x = 2.0f;
+        request->filters.bounds.points[0].y = 3.0f;
+        request->filters.bounds.points[0].z = 4.0f;
+        request->filters.bounds.points[1].x = -2.0f;
+        request->filters.bounds.points[1].y = -3.0f;
+        request->filters.bounds.points[1].z = 5.0f;
+        request->filters.bounds.type = simulation_interfaces::msg::Bounds::TYPE_BOX;
+
+        auto future = client->async_send_request(request);
+        SpinAppSome();
+
+        ASSERT_TRUE(future.wait_for(std::chrono::seconds(0)) == std::future_status::ready) << "Service call timed out.";
+        auto response = future.get();
+        EXPECT_NE(response->result.result, simulation_interfaces::msg::Result::RESULT_OK);
+    }
+
+    //! check if capabilities vector is empty when no features are provided by
+    TEST_F(SimulationInterfaceROS2TestFixture, GetSimulationFeaturesNoFeaturesProvidedByGem)
+    {
+        using ::testing::_;
+        auto node = GetRos2Node();
+
+        auto client = node->create_client<simulation_interfaces::srv::GetSimulatorFeatures>("/get_simulation_features");
+        auto request = std::make_shared<simulation_interfaces::srv::GetSimulatorFeatures::Request>();
+        auto future = client->async_send_request(request);
+        SpinAppSome();
+        ASSERT_TRUE(future.wait_for(std::chrono::seconds(0)) == std::future_status::ready) << "Service call timed out.";
+        auto response = future.get();
+        EXPECT_EQ(response->features.features.size(), 0) << "Features vector is not empty.";
```

Comment:
Can we use `empty()` here? It would be a bit cleaner
```suggestion
        EXPECT_TRUE(response->features.features.empty()) << "Features vector is not empty.";
```